### PR TITLE
Feature/cp 3430/list individual note and message

### PR DESCRIPTION
**Pull request by nf2000**
https://chillibean.atlassian.net/browse/CP-3430
https://chillibean.atlassian.net/browse/CP-3431

### CODE REVIEW

File: nims-notes-things/src/components/Form.js [hunk 1]
```diff
@@ -34,16 +30,11 @@ const Form = (props) => {
...
-        setFormEmpty((previousState) => {
-          return {
-            ...previousState,
-            [key]: true,
-          };
-        });
-        return (isValidated = false);
+      if (formData[key].length === 0) {
+        handleOnBlur(key);
+        return (isValidated = true);
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: remove brackets here

File: nims-notes-things/src/components/Form.js [hunk 2]
```diff
@@ -34,16 +30,11 @@ const Form = (props) => {
   };
 
   const validateSubmit = () => {
-    let isValidated = true;
+    let isValidated = false;
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: confused on logic here, why the change?

File: nims-notes-things/src/components/Form.js [hunk 3]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
   const handleOnSubmit = (event) => {
     event.preventDefault();
     const isValidated = validateSubmit();
-    if (isValidated) {
+    if (!isValidated) {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: why are you checking if it's not validated?

File: nims-notes-things/src/components/Form.js [hunk 4]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
...
 
-  const handleOnBlur = (value, key) => {
-    setFormEmpty((previousState) => {
-      return {
-        ...previousState,
-        [key]: value === "",
-      };
-    });
+  const handleOnBlur = (key) => {
+    if (formData[key] === "") {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: same with alex, wondering why the change here?

File: nims-notes-things/src/components/Form.js [hunk 5]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
...
+      document.getElementById(key).innerHTML = "";
+    }
   };
 
   return (
     <StyledForm onSubmit={handleOnSubmit}>
       <StyledDiv>
         <InnerForm>
-          {empty.name && <ErrorMessage>Name field is required</ErrorMessage>}
+          {notes.length === 0 ? <p> Please Enter your Note</p> : null}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: you can use `{notes.length === 0 && <p> Please Enter your Note</p>}`

File: nims-notes-things/src/components/Form.js [hunk 6]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
...
+          {<ErrorMessage id="name"></ErrorMessage>}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: no need for this to be in curly brackets

File: nims-notes-things/src/components/Form.js [hunk 7]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
...
           <Input
             name="name"
             placeholder="Please enter Your name"
             onChange={(e) => handleOnChange(e, "name")}
-            onBlur={(e) => handleOnBlur(e.target.value, "name")}
+            onBlur={(e) => handleOnBlur("name")}
             value={formData.name}
           />
-          {empty.date && <ErrorMessage>Date field is required</ErrorMessage>}
+          {<ErrorMessage id="date"> </ErrorMessage>}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: no need for this to be in curly brackets

File: nims-notes-things/src/components/Form.js [hunk 8]
```diff
@@ -52,59 +43,57 @@ const Form = (props) => {
...
             name="date"
             type="date"
             placeholder="Please enter the date"
             onChange={(e) => handleOnChange(e, "date")}
-            onBlur={(e) => handleOnBlur(e.target.value, "date")}
+            onBlur={(e) => handleOnBlur("date")}
             value={formData.date}
           />
-          {empty.note && <ErrorMessage>Note field is required</ErrorMessage>}
+          {<ErrorMessage id="note"> </ErrorMessage>}
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/Form.js]: no need for this to be in curly brackets

File: nims-notes-things/src/components/notes.js [hunk 9]
```diff
@@ -9,15 +10,17 @@ function Notes() {
 
   return (
     <div>
-      {notes.map((n) => {
+      {notes.map((note, index) => {
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: Very nice 🎆

File: nims-notes-things/src/components/notes.js [hunk 10]
```diff
@@ -9,15 +10,17 @@ function Notes() {
...
         return (
-          <div>
-            <Note>
-              <p>{n.name}</p>
-              <p>{n.date}</p>
-              <p>{n.note}</p>
-            </Note>
-          </div>
+          <Note key={note.id}>
+            <PinImage src={Pin} height={20} alt=""></PinImage>
```
Comment (damonspencerroberts) [inline nims-notes-things/src/components/notes.js]: add alt message

File: nims-notes-things/src/redux/actions/noteActions.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import * as noteActionTypes from "./noteActionTypes";
 
-export const setNote = (value) => {
+export const setNote = (value) => { // object formDataCopy from forms
```
Comment (damonspencerroberts) [inline nims-notes-things/src/redux/actions/noteActions.js:3]: can you move the comment above the line if you want it

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Very nice! Couple of small things to look at

File: nims-notes-things/src/components/Form.js (hunk 2 above)
Comment (nf2000) [reply nims-notes-things/src/components/Form.js]: Alex went through the userRef - so i've implemented that now :)

File: nims-notes-things/src/redux/actions/noteActions.js (hunk 11 above)
Comment (nf2000) [reply nims-notes-things/src/redux/actions/noteActions.js:3]: This comment is just there so that I could wrap my head around the redux process - i might keep it there so if in the future i get confused it's a reminder
